extensions/amp-story/1.0/story-analytics.js

```diff
@@ -99,28 +119,64 @@ export class StoryAnalyticsService {
           return;
         }
 
-        this.triggerEvent(AnalyticsEvent.PAGE_VISIBLE);
+        this.triggerEvent(StoryAnalyticsEvent.PAGE_VISIBLE);
 
         const pageIds = this.storeService_.get(StateProperty.PAGE_IDS);
         const pageIndex = this.storeService_.get(
           StateProperty.CURRENT_PAGE_INDEX
         );
         if (pageIndex === pageIds.length - 1) {
-          this.triggerEvent(AnalyticsEvent.LAST_PAGE_VISIBLE);
+          this.triggerEvent(StoryAnalyticsEvent.LAST_PAGE_VISIBLE);
         }
       },
       true /* callToInitialize */
     );
   }
 
   /**
-   * @param {!AnalyticsEvent} eventType
+   * @param {!StoryAnalyticsEvent} eventType
    */
   triggerEvent(eventType) {
+    this.incrementPageEventCount_(eventType);
     triggerAnalyticsEvent(
       this.element_,
       eventType,
-      this.variableService_.get()
+      this.updateDetails(eventType)
     );
   }
+
+  /**
+   * Updates event details.
+   * @param {!StoryAnalyticsEvent} eventType
+   * @visibleForTesting
+   * @return {!JsonObject}}
+   */
+  updateDetails(eventType) {
+    const details = {};
+    const vars = this.variableService_.get();
+    const pageId = vars['storyPageId'];
+
+    if (this.pageEventCount_[pageId][eventType] > 1) {
+      details.repeated = true;
+    }
+
+    let pageEventDetails = {};
+    pageEventDetails = {pageDetails: details};
```

Comment:
We want to preserve the field name `pageDetails`. Ideally we need to declare a JsonObject using `dict()`. And then merge the two objects using `jsonObjectAssign` that will be introduced by #24453